src/go/rpk/pkg/cli/cmd/redpanda/admin/selftest/selftest_test.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package selftest
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClusterStatus(t *testing.T) {
+	for _, test := range []struct {
+		name            string
+		serverResponse  string
+		runningNodes    []int
+		isUninitialized bool
+	}{
+		{
+			"Test runningNodes all running case",
+			`[
+               {
+                 "node_id": 1,
+                 "status": "running"
+               },
+               {
+                 "node_id": 0,
+                 "status": "running"
+               },
+               {
+                 "node_id": 2,
+                 "status": "running"
+               }
+            ]`,
+			[]int{0, 1, 2},
+			false,
+		},
+		{
+			"Test runningNodes some running case",
```

Comment:
[nit]: No need to add  "Test " to the test names, when the go test runs it will clearly show that is a test, e.g:

```
--- FAIL: TestClusterStatus (0.00s)
    --- FAIL: TestClusterStatus/runningNodes_all_running_case (0.00s)
        selftest_test.go:110: 
```